<launch>
	
	<arg name="guid" default=""/>
 	<arg name="ip" default="192.168.88.90"/>
  	<arg name="camera_info_url" default="file://$(find avt_vimba_camera)/calibrations/camera_fl_intrinsics.yaml"/>
 	<arg name="frame_id" default="left_optical"/>
	<arg name="trig_timestamp_topic" default=""/>
	<arg name="show_debug_prints" default="true"/>
 	
	<group>

		<node name="camera" pkg="avt_vimba_camera" type="mono_camera_node" output="screen">
			<param name="guid" value="$(arg guid)"/>
			<param name="ip" value="$(arg ip)"/>
			<param name="camera_info_url" value="$(arg camera_info_url)"/>
			<param name="frame_id" value="$(arg frame_id)"/>
			<param name="trig_timestamp_topic" value="$(arg trig_timestamp_topic)"/>
			<param name="show_debug_prints" value="$(arg show_debug_prints)"/>

			<!-- Trigger mode:
				1. Freerun
				2. SyncIn1
				3. SyncIn2
				4. SyncIn3
				5. SyncIn4
				6. FixedRate
				7. Software
			-->
			<param name="trigger_source" value="FixedRate"/>

			<!-- Acquisition mode:
				1. Continuous
				2. SingleFrame
				3. MultiFrame
				4. Recorder
			-->
			<param name="acquisition_mode" value="Continuous"/>

			<!-- Acquisition rate in fps -->
			<param name="acquisition_rate" value="12"/>

			<!-- Pixel format:
				1. Mono8
				2. Mono12
				3. Mono12Packed
				4. BayerRG8
				5. BayerRG12Packed
				6. BayerGR12
				7. RGB8Packed
				8. BGR8Packed
			-->
			<param name="pixel_format" value="BayerRG8"/>
			<!-- Exposure in us -->
			<param name="exposure" value="1000"/>
			<!-- Gain in dB -->
			<param name="gain" value="0"/>

			<!-- Auto control
				1. Off
				2. Once
				3. Continuous
			-->
			<param name="exposure_auto" value="Continuous"/>
			<param name="gain_auto" value="Continuous"/>
			<param name="whitebalance_auto" value="Continuous"/>

			<param name="binning_x" value="1"/>
			<param name="binning_y" value="1"/>
			<param name="decimation_x" value="1"/>
			<param name="decimation_y" value="1"/>
			<param name="x_offset" value="0"/>
			<param name="y_offset" value="0"/>

			<!-- TODO: Recalibrate camera and lidar due to updated image dimensions 
				 Issue #1241: https://github.com/usdot-fhwa-stol/carma-platform/issues/1241
			-->
			<param name="width" value="960"/>
			<param name="height" value="720"/>

			<param name="stream_bytes_per_second" value="45000000"/>
		</node>
	</group>
</launch>
